### ISSUE TITLE
Unexpected overriding behaviour of fixity declarations

### ISSUE THREAD

**Issue by ohad** [idris-lang/Idris2#2832]
This is a *usability* bug: the current behaviour is the 'intended' behaviour, but it is confusing.

Fixity declarations in client modules can lead to unexpected behaviour of the parser.
The easiest fix would be to add a warning when a module import overrides a fixity declaration (changing its precedence or fixity class (infixl/r, prefix).

# Steps to Reproduce
```bash
$ cat FixityAux.idr
```
```idris
module FixityAux

-- precedence is the opposite than expected We used to export these
-- functions, but deleted that code and forgot to delete the fixity
-- declarations as they're buried deep down in our module hierarchy
infixl 7 +++
infixl 6 ***
```
```bash
$ cat FixityDecls.idr
```
```idris
module FixityDecls

infixl 6 +++
infixl 7 ***

public export
(***),
(+++) : String -> String -> String
str1 +++ str2 = str1 ++ str2
str1 *** str2 = str1 ++ str2 ++ str1
```
```bash
$ cat FixityClient.idr
```
```idris
module FixityClient

import FixityDecls -- Global project conventions
import FixityAux   -- import for its highly useful auxiliary
                   -- definitions

abcb : String
abcb = "a" +++ "b" *** "c"
```
```bash
$ idris2 FixityClient.idr 
     ____    __     _         ___
    /  _/___/ /____(_)____   |__ \
    / // __  / ___/ / ___/   __/ /     Version 0.6.0-f443723f4
  _/ // /_/ / /  / (__  )   / __/      https://www.idris-lang.org
 /___/\__,_/_/  /_/____/   /____/      Type :? for help

Welcome to Idris 2.  Enjoy yourself!
FixityClient> :exec putStrLn abcb

```

# Expected Behavior
The programmer probably expected to see `abcb`.
A warning along the lines 'Module `FixityAux` overwrites fixity declaractions (***) and (+++)' would also constitute expected behaviour.

# Observed Behavior

The program outputs`abcab`. 

This can be hard to diagnose since the module hierarchy `FixityAux` is in doesn't even export `+++` and `***`, and so the programmer would need to `grep` the code-base to find these declarations. 
If the overwriting fixity declaration is inside some imported package (especially if it's inside some package managed by `pack` in a specific version that the current project uses), finding the offending fixity declarations might be extremely hard to find.

**Comment by ohad** [comment]
This is partly addressed by PR #2889, since it produces a warning about fixity declarations conflicting.

However, there ought to be a way to override a current fixity and suppress the warning. 
The reason is that infix ASCII operators are at a premium, and it's unreasonable to expect a global non-conflicting resolution across all possible code-bases. At some point clients will want to import two conflicting sets of notation, and will be stuck with heaps and heaps of fixity warnings that they cannot address.


This can be done, for example, on a per-module basis, so that if a client module imports two modules with conflicting fixity declarations, the client would need to resolve the conflict.

That's just one proposed solution though, I don't know enough about the fixity conflict warning mechanism to judge whether it's reasonable to implement.

**Comment by andrevidela** [comment]
If that's fine with everyone, I'm going to close this in favor of #2998 which provides a solution for the use-case outlined here.
